### PR TITLE
Improve selector search

### LINKED ISSUE #3536: Tags, categories and keywords - search feature is inefficient
I feel like I'm having more and more trouble finding what I'm looking for in GDevelop, and I'm supposed to be an experienced user.
The search fields, be they for events or examples, return too much noise.

Here, for instance, I want a mouse action, but can't remember the name, so I write "mouse".
The cursor action I'm looking for appears **below** the unrelated force actions.
I suppose it's because of "**MO**vement **US**ing forc**E**s", but what's the point of that? How comes such a far-fetched match comes before a full word category match?
And I can't use quotes in that field for a strict search, to filter the noise out.
image

Now, in the Examples section. I'm looking for an example with dragging.
The **first** result that actually contains my keyword is, as you see with the scrollbar, near the end of the results (about 15th position).
image
I suppose that all the results before that one use the draggable behavior and have been tagged as such, but I think the search result priority should be: title > description > invisible tags.
On a side note, this search didn't return the example I was looking for, because it's called "Move object with mouse joint". 😅 

I hope you can do something about it. 😃

### PR DESCRIPTION
**Pull request by AlexandreSi**
Resolves #3536 

Applies the same search behavior as for the example and extension stores.

demo_highlight_search_instruction

### CODE REVIEW

**Comment by AlexandreSi** [comment]
I used a Fuse client for the tags as well because I think the users will expect to have the same fuzzy behavior between all the displayed results.

File: newIDE/app/src/EventsSheet/InstructionEditor/InstructionOrObjectSelector.js [hunk 1]
```diff
@@ -145,8 +169,77 @@ export default class InstructionOrObjectSelector extends React.PureComponent<
...
+    });
   }
 
+  _search = (searchText: string) => {
+    if (searchText === '') return;
+
+    this.setState({
+      searchResults: {
+        objects: this.objectSearchApi
+          ? this.objectSearchApi.search(`'${searchText}`).map(result => ({
```
Comment (4ian) [inline newIDE/app/src/EventsSheet/InstructionEditor/InstructionOrObjectSelector.js:208]: What is this ' useful for?

File: newIDE/app/src/UI/Search/UseSearchStructuredItem.js [hunk 2]
```diff
@@ -34,16 +43,12 @@ const tuneMatchIndices = (match: SearchMatch, searchText: string) => {
     .filter(Boolean);
 };
 
-const tuneMatches = (
-  result: {| item: any, matches: SearchMatch[] |},
-  searchText: string
-) => {
-  return result.matches.map(match => ({
+export const tuneMatches = <T>(result: SearchResult<T>, searchText: string) =>
```
Comment (4ian) [inline newIDE/app/src/UI/Search/UseSearchStructuredItem.js:46]: 👍

**Review by 4ian — APPROVED** [review]

File: newIDE/app/src/EventsSheet/InstructionEditor/InstructionOrObjectSelector.js (hunk 1 above)
Comment (AlexandreSi) [reply newIDE/app/src/EventsSheet/InstructionEditor/InstructionOrObjectSelector.js:208]: I'm not sure but the results felt better : https://fusejs.io/examples.html#extended-search

**Pull request merged**